docker-setups/angular-docker.md

```diff
@@ -0,0 +1,310 @@
+# Setting Up Docker Environment for Angular Application with Nginx
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+   - [Purpose](#purpose)
+   - [Scope](#scope)
+2. [Prerequisites](#2-prerequisites)
+3. [Set up for Angular project](#3-set-up-for-angular-project)
+4. [Setting up the Docker Environment](#4-setting-up-the-docker-environment)
+   - [Dockerfile](#41-dockerfile)
+      - [Angular version-specific file path for Nginx](#set-the-correct-file-path-for-nginx)
+   - [Docker Compose](#42-docker-compose)
+   - [nginx.conf](#43-nginxconf)
+   - [.dockerignore](#44-dockerignore)
+5. [Testing the Docker Environment](#5-testing-the-docker-environment)
+   - [Using docker-compose](#build-and-run-using-docker-compose-file)
+   - [Step by step build](#or-alternatively-build-step-by-step)
+6. [Access Angular Application](#6-access-angular-application)
+7. [Sample Repository](#7-sample-repository)
+8. [Conclusion](#8-conclusion)
+   - [Benefits of Docker Setup](#benefits-of-docker-setup)
+   - [Future Enhancements](#future-enhancements)
+
+Each section is linked to its corresponding content in the document for easy navigation.
+
+## 1. Introduction
+
+### Purpose
+
+The purpose of this document is to provide a guide for setting up a Docker environment for an Angular 17 application and to host it on nginx server.
+
+### Scope
+
+- This document covers the basic setup of a Docker environment for an Angular application, including Dockerfile creation, Docker Compose setup, and database configuration.
+
+- Dockerization for this example is done on app built on Angular version 17.1.0
+
+- This document also covers Dockerization for apps built on Angular version 16 and below.
+
+[Back to top](#table-of-contents)
+
+## 2. Prerequisites
+
+Before setting up the Docker environment, ensure you have the following prerequisites:
+
+- Docker installed on your local machine.
+- An Angular application set up with necessary dependencies.
+- Basic knowledge of Docker and Docker Compose.
+
+Download links:
+
+- [Docker](https://docs.docker.com/get-docker/)
+- [Node.js](https://nodejs.org/en/download/)
+- [Angular CLI](https://angular.io/guide/setup-local)
+- [Visual Studio Code (Optional)](https://code.visualstudio.com/)
+
+[Back to top](#table-of-contents)
+
+## 3. Set up for Angular project
+
+In `package.json`, set the command to build the application with production configuration:
+
+```json
```

Comment:
Use `jsonc` to avoid the invalid red highlighting in this code block for comments.

Reply:
Used jsonc